prow/crier/handler.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package crier
+
+import (
+	"github.com/sirupsen/logrus"
+)
+
+// Handler interface contains the methods that are required
+type Handler interface {
```

Comment:
You should be writing a level-driven system for a controller. We look at current state of the `ProwJob.Status` and the current state of the report in GitHub, and reconcile as necessary. Do not try to generate an event-driven system since you are not ensured to see every event.

Reply:
I think I'll only handle `ObjectUpdated` and compare if it is a status change, and calls into reporter lib from there

Reply:
I'll take your example as I don't want to handle add/delete

Reply:
wait, 

> since you are not ensured to see every event

really? why? then this won't be useful?

Reply:
from https://github.com/kubernetes/sample-controller/blob/master/docs/controller-client-go.md - I think missing events will be handled by relisting from the list-watcher, crier should only care about the state changes and the reporter lib should handle the underneath logic. I'll verify the behavior.

Reply:
Yep, if you miss events you'll see the object on your next relist. I just mean (and this is what you said you were about to do in scrum) that you should reconcile current state with current spec. Don't try to digest into "this happened" events, just compare desired and actual state and nudge in the convergent direction